mmm_experiments/agents/exafs.py

```diff
@@ -0,0 +1,244 @@
+from collections import namedtuple
+from pathlib import Path
+from typing import Literal, Tuple, Union
+
+
+from botorch.acquisition import UpperConfidenceBound
+from botorch.acquisition.monte_carlo import qUpperConfidenceBound
+from botorch.fit import fit_gpytorch_model
+from botorch.models import SingleTaskGP
+from botorch.optim import optimize_acqf
+from databroker.client import BlueskyRun
+from gpytorch.mlls import ExactMarginalLogLikelihood
+import numpy as np
+import torch
+
+from .base import Agent
+
+DATA_KEY = None  # TODO
+
+
+class DumbDistanceEXAFSAgent(Agent):
+    """The point of this agent is to do precisely one thing: maximize the
+    distance between some newly acquired point(s) and the provided reference
+    spectra. The reference EXAFS represent "known phases". The argument is
+    simple: if a new EXAFS has a e.g. Euclidean distance that is large compared
+    to the reference, it is likely "different" than the reference. If it has
+    a distance that is small, it is likely not different than the reference.
+    Multiple references can be provided, and the distance considered will be
+    the minimum distance for any of the references."""
+
+    server_host = None  # TODO "qserver1.nsls2.bnl.gov:60610"
+    measurement_plan_name = None  # TODO ? "mv_and_jog"
+
+    def __init__(
+        self,
+        sample_origin,
+        relative_bounds,
+        reference_spectra,
+        device="cpu",
+        beta_UCB=0.1,
+    ):
+        """
+        Parameters
+        ----------
+        sample_origin : float
+            Decided origin of sample in raw motor coordinates. Assumed to be
+            one dimensional for now. TODO
+        relative_bounds : tuple
+            Relative bounds for the sample measurement. For a 10 cm sample,
+            this would be something like (1, 99).
+        reference_spectra : array_like
+            A list of lists or array where the first axis is the spectrum
+            index. These are the references that we're comparing measured
+            spectra against.
+        device : str
+            Torch device to keep the GP. This model should be quite cheap so
+            by default we'll put it on the cpu.
+        beta_UCB : float, optional
+            Default value for the beta parameter in the Upper Confidence Bound
+            acquisition function.
+        """
+
+        AVAIL_DEVICES = ["cpu", "cuda:0", "cuda:1", "cuda:2", "cuda:3"]
+        if device not in AVAIL_DEVICES:
+            raise ValueError(f"device={device} must be one of {AVAIL_DEVICES}")
+
+        if len(relative_bounds) != 2:
+            raise ValueError(
+                f"relative_bounds={relative_bounds} must be a length-2 tuple "
+                "of floats"
+            )
+
+        super().__init__(beamline_tla="exafs")  # <- is this correct? TODO
```

Comment:
```suggestion
        super().__init__(beamline_tla="bmm") 
```

TLA stands for three letter acronym. This is used for standardizing addresses for kafka, data bases, etc.